rust/lancedb/src/connection.rs

```diff
@@ -577,6 +633,30 @@ impl Connection {
         )
     }
 
+    /// Clone a table in the database
+    ///
+    /// Creates a new table by cloning from an existing source table.
+    /// By default, this performs a shallow clone where the new table shares
+    /// the underlying data files with the source table.
+    ///
+    /// # Parameters
+    /// - `target_table_name`: The name of the new table to create
+    /// - `source_uri`: The URI of the source table to clone from
```

Comment:
Can you elaborate more on why we are accepting a `source_uri` instead of `source_table_name`? It seems other APIs all use `name`.